src/Angular.js

```diff
@@ -399,6 +399,11 @@ function toInt(str) {
   return parseInt(str, 10);
 }
 
+var _isNaN = window.isNaN;
+var isNumberNaN = Number.isNaN || function isNumberNaN(num) {
+  return isNumber(num) ? _isNaN(num) : false;
```

Comment:
This could be just `isNumber(num) && _isNaN(num)`. Or what @realityking said probably even better...